k-distribution/tests/regression-new/issue-1633/lexical.k

```diff
@@ -0,0 +1,62 @@
+module LEXICAL-SORTS
+  syntax Identifier [token]
+  syntax IntConstant [token]
+  syntax FloatConstant [token]
+  syntax CharConstant [token]
+  syntax StringLiteral [token]
+endmodule
+
+module LEXICAL-SYNTAX
+  imports LEXICAL-SORTS
+
+  syntax Identifier ::= r"{IdentifierNonDigit}(({IdentifierNonDigit}|{Digit})*)" [token]
+  syntax lexical IdentifierNonDigit = r"{Nondigit}|{UniversalCharacterName}"
+  syntax lexical Nondigit = r"[_a-zA-Z]"
+  syntax lexical Digit = r"[0-9]"
+
+  syntax lexical UniversalCharacterName = r"(\\\\u{HexQuad})|(\\\\U{HexQuad}{2})"
+  syntax lexical HexQuad = r"{HexDigit}{4}"
+
+  syntax IntConstant ::= r"({DecConstant}|{OctConstant}|{HexConstant})({IntSuffix}?)" [token]
+  syntax lexical DecConstant = r"{NonzeroDigit}({Digit}*)"
+  syntax lexical OctConstant = r"0({OctDigit}*)"
+  syntax lexical HexConstant = r"{HexPrefix}({HexDigit}+)"
+  syntax lexical HexPrefix = r"0x|0X"
+  syntax lexical NonzeroDigit = r"[1-9]"
+  syntax lexical OctDigit = r"[0-7]"
```

Comment:
Can these have attributes? If so, can you just put some dummy attributes on some of these to demonstrate that it parses correctly?

Reply:
They cannot currently have attributes, similar to some of the other auxiliary syntax sentences like `syntax priorities`. We can change this if we want, but right now no attributes on such a sentence have any meaning whatsoever, so it seems premature.